lib/ansible/plugins/callback/__init__.py

```diff
@@ -95,6 +95,9 @@ def _dump_results(self, result, indent=None, sort_keys=True, keep_invocation=Fal
         if 'exception' in abridged_result:
             del abridged_result['exception']
 
+        # ensure we don't hit decode errors on displaying errors
+        abridged_result = to_text(abridged_result)
+
```

Comment:
sorry i should have been more explicit, we dont really want to modify result itself if possible
```
   abridged_result = to_text(strip_internal_keys(result))
```